packages/entities/entities-plugins/src/components/PluginForm.cy.ts

```diff
@@ -0,0 +1,747 @@
+// Cypress component test spec file
+import type { Router } from 'vue-router'
+import { createMemoryHistory, createRouter } from 'vue-router'
+import { EntityBaseForm } from '@kong-ui-public/entities-shared'
+import { type KongManagerPluginFormConfig, type KonnectPluginFormConfig } from '../types'
+import {
+  schema,
+  credentialSchema,
+  plugin1,
+  schema2,
+  scopedService,
+  scopedConsumer,
+} from '../../fixtures/mockData'
+import PluginForm from './PluginForm.vue'
+
+const baseConfigKonnect: KonnectPluginFormConfig = {
+  app: 'konnect',
+  apiBaseUrl: '/us/kong-api',
+  controlPlaneId: 'abc-123-i-love-cats',
+  backRoute: { name: 'select-plugin' },
+  cancelRoute: { name: 'home' },
+}
+
+const baseConfigKM:KongManagerPluginFormConfig = {
+  app: 'kongManager',
+  workspace: 'default',
+  apiBaseUrl: '/kong-manager',
+  backRoute: { name: 'select-plugin' },
+  cancelRoute: { name: 'home' },
+}
+
+describe('<PluginForm />', () => {
+  describe('Kong Manager', () => {
+    const interceptKM = (params?: {
+      mockData?: object
+      alias?: string
+    }) => {
+      cy.intercept(
+        {
+          method: 'GET',
+          url: `${baseConfigKM.apiBaseUrl}/${baseConfigKM.workspace}/schemas/plugins/*`,
+        },
+        {
+          statusCode: 200,
+          body: params?.mockData ?? schema,
+        },
+      ).as(params?.alias ?? 'getPluginSchema')
+    }
+
+    const interceptKMCreatePlugin = (params?: {
+      mockData?: object
+      alias?: string
+      status?: number
+      entityType?: string,
+      entityId?: string,
+    }): void => {
+      const url = params?.entityId
+        ? `${baseConfigKM.apiBaseUrl}/${baseConfigKM.workspace}/${params?.entityType}/${params.entityId}/plugins`
+        : `${baseConfigKM.apiBaseUrl}/${baseConfigKM.workspace}/plugins`
+
+      cy.intercept(
+        {
+          method: 'POST',
+          url,
+        },
+        {
+          statusCode: params?.status ?? 200,
+          body: params?.mockData ?? plugin1,
+        },
+      ).as(params?.alias ?? 'createPlugin')
+    }
+
+    const interceptKMOperatePlugin = (params: {
+      method: 'GET' | 'PATCH',
+      mockData?: object
+      alias?: string
+      status?: number
+      entityType?: string,
+      entityId?: string,
+    }) => {
+      const url = params?.entityId
+        ? `${baseConfigKM.apiBaseUrl}/${baseConfigKM.workspace}/${params.entityType}/${params.entityId}/plugins/*`
+        : `${baseConfigKM.apiBaseUrl}/${baseConfigKM.workspace}/plugins/*`
+
+      cy.intercept(
+        {
+          method: params.method,
+          url,
+        },
+        {
+          statusCode: 200,
+          body: params.mockData ?? plugin1,
+        },
+      ).as(params?.alias ?? 'operatePlugin')
+    }
+
+    it('should show the select page', () => {
+      interceptKM()
+
+      cy.mount(PluginForm, {
+        props: {
+          config: baseConfigKM,
+        },
+      })
```

Comment:
I believe when you mount the component you need to register the global components that are needed with it, so e.g.
```suggestion
import { VueFormGenerator } from '{path-to-source}'

// ...more code

      cy.mount(PluginForm, {
        global: {
          components: { VueFormGenerator },
        },
        props: {
          config: baseConfigKM,
        },
      })
```